relay-metrics/src/aggregation.rs

```diff
@@ -1142,11 +1149,13 @@ impl Aggregator {
     /// Pop and return the buckets that are eligible for flushing out according to bucket interval.
     ///
     /// Note that this function is primarily intended for tests.
-    pub fn pop_flush_buckets(&mut self, force: bool) -> HashMap<ProjectKey, Vec<Bucket>> {
+    pub fn pop_flush_buckets(&mut self) -> HashMap<ProjectKey, Vec<Bucket>> {
         relay_statsd::metric!(gauge(MetricGauges::Buckets) = self.buckets.len() as u64);
 
         let mut buckets = HashMap::<ProjectKey, Vec<Bucket>>::new();
 
+        let force = matches!(&self.state, AggregatorState::ShuttingDown);
+
         relay_statsd::metric!(timer(MetricTimers::BucketsScanDuration), {
             let bucket_interval = self.config.bucket_interval;
             self.buckets.retain(|key, entry| {
```

Comment:
@jan-auer I did not refactor this into a force and non-force branch after all. Since we need to iterate through `self.buckets` in order to construct the return value, I did not see much benefit.